### ISSUE TITLE
Throws KeyError: 0 with `mkdocs serve` when pages key is present

### ISSUE THREAD

**Issue by agconti** [mkdocs/mkdocs#594]
Following the [`mkdocs.yml`](https://github.com/tomchristie/django-rest-framework/blob/8935db1be92a2e92860f57cd01ebf11e09a08e9d/mkdocs.yml#L10) file in Django Rest-framework I've configured my `mkdocs.yml` file to: 

```
site_name: piedpiper
site_description: Its all about a Weissman score > 5.0
repo_url: https://github.com/agconti/piedpiper-web
site_dir: site
copyright: Copyright &copy; 2015, <a href="https://github.com/agconti">agconti</a>.
pages:
    - Home: 'index.md'
    - About: 'about.md'
```
When I call mkdocs serve or build  it throws `KeyError: 0`. When I delete the pages key it works and shows both the index, the about page, and live reloads accordingly.

New mkdocs.yml file ( that works ):
``` 
site_name: piedpiper
site_description: Its all about a Weissman score > 5.0
repo_url: https://github.com/agconti/piedpiper-web
site_dir: site
copyright: Copyright &copy; 2015, <a href="https://github.com/agconti">agconti</a>.
```
The first example with the pages key is identical to the example in the [docs](http://www.mkdocs.org/user-guide/writing-your-docs/#configure-pages-and-navigation). Any idea what might be going on? 

full stack trace: 
```bash 
 File "/usr/local/bin/mkdocs", line 9, in <module>
    load_entry_point('mkdocs==0.12.2', 'console_scripts', 'mkdocs')()
  File "/usr/local/lib/python2.7/site-packages/mkdocs/main.py", line 77, in run_main
    main(cmd, args=sys.argv[2:], options=dict(opts))
  File "/usr/local/lib/python2.7/site-packages/mkdocs/main.py", line 57, in main
    config = load_config(options=options)
  File "/usr/local/lib/python2.7/site-packages/mkdocs/config.py", line 92, in load_config
    return validate_config(user_config)
  File "/usr/local/lib/python2.7/site-packages/mkdocs/config.py", line 151, in validate_config
    path = page_config[0]
KeyError: 0
```
Using:
```
mkdocs==0.12.2
```

**Comment by d0ugal** [comment]
Django Rest Framework is using MkDocs 0.13.3, you'll need to update to use the new Pages configuration syntax.

http://www.mkdocs.org/about/release-notes/#change-the-pages-configuration

**Comment by agconti** [comment]
@d0ugal thanks! I thought I was on the most recent version. I apologize.

**Comment by d0ugal** [comment]
@agconti No problem, we only just recently released 0.13. A 0.14 release is going to follow soon due to problems for some users in 0.13.
